channeldb/invoices.go

```diff
@@ -379,7 +380,7 @@ func deserializeInvoice(r io.Reader) (*Invoice, error) {
 	invoice.Terms.Value = lnwire.MilliSatoshi(byteOrder.Uint64(scratch[:]))
 
 	var settleByte [1]byte
-	if _, err := io.ReadFull(r, settleByte[:]); err != nil {
+	if err := binary.Read(r, byteOrder, settleByte[:]); err != nil {
```

Comment:
We should be able to pass `i.Terms.Settled` directly into the `binary.Write` here, eliminating the declaration of `settleByte` all together.